notification/interfaces.go

```diff
@@ -0,0 +1,7 @@
+package notification
+
+import "context"
+
+type Provider interface {
+	Notifications(ctx context.Context, filter *Filter) ([]Result, error)
```

Comment:
It would be great if you can provide this function documentation. You can refer to <https://tip.golang.org/doc/comment>

Reply:
hai, i added the comments